templates/admin/src/app.raw.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from "react";
+import { Theme, withStyles } from "@material-ui/core/styles";
+
+import GenericApp from "@iobroker/adapter-react/GenericApp";
+import Settings from "./components/settings";
+import { GenericAppProps, GenericAppSettings } from "@iobroker/adapter-react/types";
+import { StyleRules } from "@material-ui/styles";
+
+const styles = (_theme: Theme): StyleRules => ({
+	root: {},
+});
+
+class App extends GenericApp {
+	constructor(props: GenericAppProps) {
+		const extendedProps: GenericAppSettings = { ...props };
+		extendedProps.encryptedFields = [];
+		extendedProps.translations = {
+			"en": require("./i18n/en"),
+			"de": require("./i18n/de"),
+			"ru": require("./i18n/ru"),
+			"pt": require("./i18n/pt"),
+			"nl": require("./i18n/nl"),
+			"fr": require("./i18n/fr"),
+			"it": require("./i18n/it"),
+			"es": require("./i18n/es"),
+			"pl": require("./i18n/pl"),
+			"zh-cn": require("./i18n/zh-cn"),
+		};
+
+		super(props, extendedProps);
+	}
```

Comment:
How would it look like when a dev wants to add custom properties to `this.state`? I haven't used that in combination with inheritance yet. An example (commented out) would probably be good.

Reply:
Currently the state of the `App` can't be extended (at least not easily). I believe there is no need to have additional state in `App`. Everything should be done inside the `<Settings>` component; there you can define whatever you like as state object. `<App>` is basically the outer "frame" of the settings `iframe` which contains buttons and message dialogs.

Reply:
Ok makes sense.